src/workerd/api/sql.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <workerd/jsg/jsg.h>
+#include <workerd/jsg/string.h>
+#include "basics.h"
+
+namespace workerd::api {
+
+typedef kj::OneOf<kj::Array<const byte>, kj::String, double> ValueBind;
+class SqlDatabase;
+
+class SqlResult final: public jsg::Object {
+public:
+  SqlResult(SqliteDatabase::Query& query);
+
+  kj::ArrayPtr<jsg::Dict<SqliteDatabase::Query::ValueOwned>> getResults(jsg::Lock& js);
+
+  JSG_RESOURCE_TYPE(SqlResult, CompatibilityFlags::Reader flags) {
+    JSG_READONLY_PROTOTYPE_PROPERTY(results, getResults);
+  }
+
+private:
+  kj::Vector<jsg::Dict<SqliteDatabase::Query::ValueOwned>> results;
+};
+
+class SqlPreparedStatement final: public jsg::Object {
+public:
+  SqlPreparedStatement(jsg::Ref<SqlDatabase>&& sqlDb, SqliteDatabase::Statement&& statement);
+
+  jsg::Ref<SqlResult> run(kj::Maybe<kj::Array<ValueBind>> bindValuesOptional);
+
+  JSG_RESOURCE_TYPE(SqlPreparedStatement, CompatibilityFlags::Reader flags) {
+    JSG_METHOD(run);
+  }
+
+  void visitForGc(jsg::GcVisitor& visitor) {
+    visitor.visit(sqlDatabase);
+  }
+
+private:
+  jsg::Ref<SqlDatabase> sqlDatabase;
+  SqliteDatabase::Statement statement;
+};
+
+class SqlDatabase final: public jsg::Object {
+public:
+  friend class SqlPreparedStatement;
+  SqlDatabase();
+
+  static jsg::Ref<SqlDatabase> constructor(jsg::Lock& js, CompatibilityFlags::Reader flags);
```

Comment:
So, this appears to create an in-memory database. But that's not what we want, right? We want to use the database that's already being opened for a Durable Object, and is actually durable.

I would expect the sqlite storage stuff to hang off actor storage, like instead of `storage.get(key)` now you can do `storage.sqlite.run(statement)`.

Reply:
We can certainly switch that over, I think the short term idea was just to have something workable that we can test with (if we use storage.sqlite, will that actually write to the disk?)
But happy to move it over